requirements: Relax protobuf version constraint

juergbi opened the pull request:
All generated code since 3.20.0 will be supported until at least 8.x.y.

https://protobuf.dev/support/cross-version-runtime-guarantee/
https://github.com/protocolbuffers/protobuf/commit/1af7fd407c85c9c1f34b6ca8441cfb3a72d4f5ea

juergbi committed:
requirements: Relax protobuf version constraint

All generated code since 3.20.0 will be supported until at least 8.x.y.

https://protobuf.dev/support/cross-version-runtime-guarantee/
https://github.com/protocolbuffers/protobuf/commit/1af7fd407c85c9c1f34b6ca8441cfb3a72d4f5ea

abderrahim commented:
I would revert https://github.com/apache/buildstream/pull/2017 wholesale, but given this new info it would be nice to have your changes on top of that.

juergbi committed:
Add tox environment `build-grpc`

We need to make sure that the right version of `grpcio-tools` is used to
regenerate protobuf and grpc code to not depend on too recent versions
of protobuf or grpc. Define a tox environment to make this more
convenient and less error-prone.

juergbi committed:
Revert "Add note to update the requirements whenever the grpc code is regenerated"

This reverts commit 5e9eb76017fc875461ee86f3ff6f398be3ee0b7f.

juergbi commented:
I've reverted the remaining part of #2017. And I've added a tox environment `build-grpc` that regenerates the code with the desired version of `grpcio-tools` with a small documentation update.
